src/pudl_archiver/__init__.py

```diff
@@ -68,23 +48,24 @@ async def archive_datasets(
         # List to gather all archivers to run asyncronously
         tasks = []
         for dataset in datasets:
-            zenodo_client = ZenodoClient(
-                "dataset_doi.yaml",
+            cls = ARCHIVERS.get(dataset)
+            if not cls:
+                raise RuntimeError(f"Dataset {dataset} not supported")
+            else:
+                downloader = cls(session)
+            zenodo_deposition_interface = ZenodoDepositionInterface(
+                dataset,
+                downloader,
```

Comment:
Now we pass the downloader into `ZDI`, so all the coordination lives there. Name-wise it doesn't quite make sense with `AbstractDatasetArchiver` probably being a good place to run the orchestration - but I think what *does* make sense is having the "downloader" and the "depositor" both being controlled by some higher being.